action/candidateregister_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2020 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package action
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCandidateRegister(t *testing.T) {
+	require := require.New(t)
+	cr, err := NewCandidateRegister(nonce, canAddress, canAddress, canAddress, canAddress, amount.Text(10), duration, autoStake, payload, gaslimit, gasprice)
+	require.NoError(err)
+
+	ser := cr.Serialize()
+	require.Equal("0a81010a29696f3178707136326177383575717a72636367397935686e727976386c64326e6b7079636333677a611229696f3178707136326177383575717a72636367397935686e727976386c64326e6b7079636333677a611a29696f3178707136326177383575717a72636367397935686e727976386c64326e6b7079636333677a611202313018e80720012a29696f3178707136326177383575717a72636367397935686e727976386c64326e6b7079636333677a6132077061796c6f6164", hex.EncodeToString(ser))
+
+	require.NoError(err)
+	require.Equal(gaslimit, cr.GasLimit())
+	require.Equal(gasprice, cr.GasPrice())
+	require.Equal(nonce, cr.Nonce())
+
+	require.Equal(canAddress, cr.Name())
+	require.Equal(canAddress, cr.OperatorAddress().String())
+	require.Equal(canAddress, cr.RewardAddress().String())
+	require.Equal(canAddress, cr.OwnerAddress().String())
+	require.Equal(amount, cr.Amount())
+	require.Equal(duration, cr.Duration())
+	require.True(cr.AutoStake())
+	require.Equal(payload, cr.Payload())
+
+	gas, err := cr.IntrinsicGas()
+	require.NoError(err)
+	require.Equal(uint64(10700), gas)
+	cost, err := cr.Cost()
+	require.NoError(err)
+	require.Equal("107010", cost.Text(10))
+
+	proto := cr.Proto()
+	cr2 := &CandidateRegister{}
+	require.NoError(cr2.LoadProto(proto))
+	require.Equal(canAddress, cr2.Name())
```

Comment:
can simplify to Equal(cr, cr2)

Reply:
not equal because of nonce gas and gas price